Common/Core/aodMerger.cxx

```diff
@@ -193,14 +194,42 @@ int main(int argc, char* argv[])
       ++totalMergedDFs;
       auto folder = (TDirectoryFile*)inputFile->Get(dfName);
       auto treeList = folder->GetListOfKeys();
+
+      treeList->Sort();
+
+      // purging keys from duplicates
+      for (auto i = 0; i < treeList->GetEntries(); ++i) {
+        TKey* ki = (TKey*)treeList->At(i);
+        for (int j = i + 1; j < treeList->GetEntries(); ++j) {
+          TKey* kj = (TKey*)treeList->At(j);
+          if (std::strcmp(ki->GetName(), kj->GetName()) == 0 && std::strcmp(ki->GetTitle(), kj->GetTitle()) == 0) {
+            if (ki->GetCycle() < kj->GetCycle()) {
+              printf("    ***WARNING*** we had ordered the keys, first cycle should be higher, let's continue anyway");
+            } else {
+              // key is a duplicate, let's remove it
+              treeList->Remove(kj);
```

Comment:
Given that you manipulate the list in-place while you iterate it, don't you need to add `j--` here to make sure the subsequent entry (which becomes the current on removal) is checked as well?

Reply:
Yes, of course, thanks for spotting.